### ISSUE TITLE
Concerns about the removal of batch manager C++ API

### ISSUE THREAD

**Issue by DreamGenX** [NVIDIA/TensorRT-LLM#2114]
### System Info

Hello, in the [latest release](https://github.com/NVIDIA/TensorRT-LLM/pull/2110) you say:

> The C++ batch manager API is deprecated in favor of the C++ executor API, and it will be removed in a future release of TensorRT-LLM.

However, the C++ executor API is not as flexible -- for example, it's not possible to implement custom token sampling algorithms such as MinP or structured decoding parametrized by user provided schema or grammar. See here for details: https://github.com/NVIDIA/TensorRT-LLM/issues/2044

Can you please share your timeline for removal and whether you plan to bring the C++ executor API on par with the batch manager?

### Who can help?

@bys

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

N/A

### Expected behavior

N/A

### actual behavior

N/A

### additional notes

N/A

**Comment by Bam4d** [comment]
I also am very concerned about this change, which would be easier to deal with if the executor API allowed you to set the requestId, which used to be possible in the batchmanager API

The fact that you cannot set this means it is hard to keep track of which requests match which responses, without keeping a distributed map across MPI if you are using multiple processes ... which is a huge amount of work.

We would then also have to additionally set up this mapping across all of MPI nodes for the logits post-processing etc, so we can look up request details.

**Comment by DreamGenX** [comment]
It looks like the executor API now lets you specify [clientId](https://github.com/NVIDIA/TensorRT-LLM/blob/b8fc6633ba27550dea5f33641f5d94b6c7f02125/cpp/include/tensorrt_llm/executor/executor.h#L395) that is then passed to the logit post processor as well.

**Comment by amukkara** [comment]
@Bam4d Logits processor runs only on one MPI node in latest version.
you can use LogitsPostProcessorConfig(replicate=True) if you want processor to run all tensor parallel ranks.

**Comment by akhoroshev** [comment]
> Logits processor runs only on one MPI node in latest version.
> you can use LogitsPostProcessorConfig(replicate=True) if you want processor to run all tensor parallel ranks.

If I change logits at rank 0 in the logit processor, do I need to do the same logit transformation at other ranks? https://github.com/NVIDIA/TensorRT-LLM/issues/1415

**Comment by amukkara** [comment]
there is no need for logits transformation on other ranks in client code.

logits callback will be invoked only in first tensor parallel rank. A broadcast will be performed in executor to keep all tensor parallel ranks consistent.

**Comment by karljang** [comment]
Closing the issue as stale, please feel free to open a new one, if the issue is still valid in the latest release.
Thank you!
